### ISSUE TITLE
stderr: ERROR: Unable to locate a modulefile for 'PrgEnv-pgi' 

### ISSUE THREAD

**Issue by menzzana** [easybuilders/easybuild-easyblocks#2283]
I am installing CrayCCE on the a Cray XC40 and am recieving the error.

```
stderr: ERROR: Unable to locate a modulefile for 'PrgEnv-pgi'\n)") 
```

Looking at the LOG I see...

```
module unload PrgEnv-gnu
module unload PrgEnv-intel
module unload PrgEnv-pgi

; stderr: ERROR: Unable to locate a modulefile for 'PrgEnv-pgi' 
```

In our system we do not even have `PrgEnv-pgi` so it is natural that we get this error, but we are not using pgi either so
it should be checked if such a module exists prior of unloading it.
Also `PrgEnv-pgi` is not listed as a dependency in the config file.

We are using module.tcl Modules Release 4.1.3.1 (2018-11-29)
and I know that EasyBuild prefers Lmod, but still module.tcl is supported.

Cheers
Henric

**Comment by boegel** [comment]
Thanks for reporting this @menzzana !

Usually a modules tool is happy to unload a module that isn't loaded, even if the module file doesn't exist, but apparently 4.1.3 doesn't... Maybe @xdelaruelle can shed some light on this?

In any case, it should be fairly easy to fix this, by guarding the `module unload` with a condition.
That change should be made in the [CrayToolchain](https://github.com/easybuilders/easybuild-easyblocks/blob/develop/easybuild/easyblocks/generic/craytoolchain.py)

**Comment by boegel** [comment]
@menzzana The changes in https://github.com/easybuilders/easybuild-easyblocks/pull/2286 should fix the problem you're seeing.

You can try this yourself using "`eb --include-easyblocks-from-pr 2286`" when installing the `CrayCCE` toolchain.

**Comment by xdelaruelle** [comment]
@menzzana @boegel  The issue encountered is due to a behavior change introduced on Modules version 4.0. Starting Modules 4.2 the original behavior was restored (unloading an unexistent modulefile does not produce an error). This issue was also reported on cea-hpc/modules#199.

**Comment by menzzana** [comment]
Thanks for the advice and solution. I included the extra flag and then it worked perfectly.
I will leave the question open so you can decide if you want to close the issue or keep it open as reminder
for future update

**Comment by boegel** [comment]
This issue will get closed when the fix in #2286 gets merged (which means the change will make it in the next EasyBuild release).
